analysers/trials/numberOfProcessInstances.py

```diff
@@ -23,10 +24,95 @@
     .set("spark.cassandra.connection.host", cassandraHost)
 sc = CassandraSparkContext(conf=conf)
 
-data = sc.cassandraTable(cassandraKeyspace, srcTable)\
-        .select("duration") \
+dataRDD = sc.cassandraTable(cassandraKeyspace, srcTable)\
+        .select("process_definition_id", "source_process_instance_id", "end_time", "start_time", "duration") \
         .where("trial_id=? AND experiment_id=?", trialID, experimentID) \
-        .map(lambda r: 1) \
+        .filter(lambda r: r["process_definition_id"] is not None) \
+        .cache()
+
+processes = dataRDD.map(lambda r: r["process_definition_id"]) \
+        .distinct() \
+        .collect()
+
+maxTime = None
+maxID = None
+for p in processes:
+    time = dataRDD.filter(lambda r: r["process_definition_id"] == p) \
+        .map(lambda r: (r["start_time"], r["source_process_instance_id"])) \
+        .sortByKey(1, 1) \
+        .collect()
+    time = time[nToIgnore-1]
+    if maxTime is None or time[0] > maxTime:
+        maxTime = time[0]
+        maxID = time[1]
+
+print(maxTime)
+
+data = dataRDD.map(lambda r: (r["start_time"], r)) \
+        .sortByKey(1, 1) \
+        .map(lambda r: r[1]) \
+        .collect()
+
+index = 0
+for i in range(len(data)):
+    if data[i]["source_process_instance_id"] == maxID:
+        index = i
+        break
+
+data = sc.parallelize(data[index:])
+
+data = data.map(lambda r: 1) \
+        .reduce(lambda a, b: a+b)
+
+query = [{"experiment_id":experimentID, "trial_id":trialID, "number_of_process_instances":data}]
+
+sc.parallelize(query).saveToCassandra(cassandraKeyspace, destTable)
+
+print(data)# Set configuration for spark context
```

Comment:
@VincenzoFerme Yeah I could just use the length of the data slice straight away.